File: caff-browser-backend/src/main/java/hu/bme/hit/crysis/sludgeeldiablo/caffbrowser/dto/ImageDto.java
package hu.bme.hit.crysis.sludgeeldiablo.caffbrowser.dto;

import lombok.Data;

import java.time.LocalDateTime;
import java.util.List;
import java.util.Set;

@Data
public class ImageDto {

    private Long id;
    private String userDisplayName;
    private String gifPath;
    private String caffPath;
    private List<CommentDto> comments;
    private Integer commentsSize;
    private Boolean commentable;
    private Boolean modifiable;

    // CAFF meta data
    private LocalDateTime date;
    private String credit;
    private String caption;
    private Set<String> tags;
    private Long width;
    private Long height;
}

File: caff-browser-frontend/src/services/openapi/models/ImageDto.ts
/* istanbul ignore file */
/* tslint:disable */
/* eslint-disable */

import type { CommentDto } from './CommentDto';

export type ImageDto = {
    id?: number;
    userDisplayName?: string;
    gifPath?: string;
    caffPath?: string;
    comments?: Array<CommentDto>;
    commentsSize?: number;
    commentable?: boolean;
    modifiable?: boolean;
    date?: string;
    credit?: string;
    caption?: string;
    tags?: Array<string>;
    width?: number;
    height?: number;
}